### PR TITLE
Do not allow statement linear address to be zero

### PR DESCRIPTION
**Pull request by ptomin**
Do not allow statement linear address to be zero: throw `ArgumentException`

### CODE REVIEW

**Comment by ptomin** [comment]
63 unit tests failed...

**Comment by ptomin** [comment]
47 unit tests failed...

**Comment by ptomin** [comment]
38 unit tests failed...

**Comment by ptomin** [comment]
37 unit tests failed...

**Comment by ptomin** [comment]
31 unit tests failed...

**Comment by ptomin** [comment]
28 unit tests failed...

**Comment by ptomin** [comment]
4 unit tests failed...

**Comment by ptomin** [comment]
2 unit tests failed...

**Comment by ptomin** [comment]
There are `zero` addresses in `PIC18`, `TLCS-90`, `i8051`. It seems to be correct, is not it, @uxmal?
So `zero`-check should be removed from `Statement` constructor

**Comment by uxmal** [comment]
Indeed. There are many CPU's where address 0 is perfectly legal (Z80, PDP-11, x86 real mode come to mind).

**Comment by uxmal** [comment]
Let's go with this for now. I will make some changes to ```Procedure.EntryAddress``` following this PR.

File: src/Decompiler/Analysis/LiveCopyInserter.cs [hunk 1]
```diff
@@ -55,7 +55,10 @@ public int IndexOfInsertedCopy(Block b)
 
 		public Identifier InsertAssignmentNewId(Identifier idOld, Block b, int i)
 		{
-			Statement stm = new Statement(0, null, b);
+            var stm = new Statement(
+                (b.Address ?? b.Procedure.EntryAddress).ToLinear(),
```
[the remark this answers is not in the document]
Comment (ptomin) [reply src/Decompiler/Analysis/LiveCopyInserter.cs]: Entry block, exit block, maybe some synthesized blocks in scanner, e.g.
https://github.com/uxmal/reko/blob/010a71ee32e8e53e338b43914d88be5f25ee3544/src/Decompiler/Scanning/Scanner.cs#L424

**Pull request merged**